mlflow/utils/rest_utils.py

```diff
@@ -14,26 +12,76 @@
 from mlflow.utils.string_utils import strip_suffix
 from mlflow.exceptions import MlflowException, RestException
 
-_REST_API_PATH_PREFIX = "/api/2.0"
 RESOURCE_DOES_NOT_EXIST = "RESOURCE_DOES_NOT_EXIST"
+_REST_API_PATH_PREFIX = "/api/2.0"
+_DEFAULT_HEADERS = {"User-Agent": "mlflow-python-client/%s" % __version__}
+# Response codes that generally indicate transient network failures and merit client retries,
+# based on guidance from cloud service providers
+# (https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#general-rest-and-retry-guidelines)
+_TRANSIENT_FAILURE_RESPONSE_CODES = [
+    408,  # Request Timeout
+    429,  # Too Many Requests
+    500,  # Internal Server Error
+    502,  # Bad Gateway
+    503,  # Service Unavailable
+    504,  # Gateway Timeout
+]
 
-_logger = logging.getLogger(__name__)
 
-_DEFAULT_HEADERS = {"User-Agent": "mlflow-python-client/%s" % __version__}
+def _get_http_response_with_retries(max_retries, backoff_factor, retry_codes, **kwargs):
+    """
+    Performs an HTTP request using Python's `requests` module with an automatic retry policy.
+
+    :max_retries: Maximum total number of retries.
```

Comment:
```suggestion
    :param max_retries: Maximum total number of retries.
```

Nit: can we add `param` to parameter descriptions?